File: 백준/Silver/5525. IOIOI/IOIOI.js
const filePath = process.platform === "linux" ? "/dev/stdin" : "./5525.txt";
let input = require("fs").readFileSync(filePath).toString().trim().split("\n");

const N = parseInt(input[0]);
const M = parseInt(input[1]);
const S = input[2];

const PN = "I" + "OI".repeat(N);

let cnt = 0;
let idx = 0;

while (true) {
  if (S.substring(idx, idx + PN.length) === PN) {
    cnt++;
    idx += 2;
  } else {
    idx++;
  }

  if (idx + PN.length > M) break;
}


console.log(cnt);

File: 백준/Silver/5525. IOIOI/IOIOI.py
import sys

input = sys.stdin.readline


N = int(input())
M = int(input())
S = input().rstrip()


cnt = 0
idx = 0
ans = 0

while idx < M - 2:

    if S[idx : idx + 3] == "IOI":
        cnt += 1
        if cnt >= N:
            ans += 1
        idx += 2
    else:
        cnt = 0
        idx += 1


print(ans)
